priv/perf/apps/load_test/lib/service/faucet.ex

```diff
@@ -45,48 +69,124 @@ defmodule LoadTest.Service.Faucet do
+      %Utxo{amount: change, currency: currency, owner: faucet_account.addr},
+      %Utxo{amount: amount, currency: currency, owner: receiver.addr}
     ]
 
-    {:ok, blknum, txindex} = Transaction.submit_tx([utxo], outputs, [faucet], @fund_child_chain_account_retries)
-    {:ok, change_utxo} = Utxo.new(%{blknum: blknum, txindex: txindex, oindex: 0})
-    {:ok, user_utxo} = Utxo.new(%{blknum: blknum, txindex: txindex, oindex: 1, amount: amount})
+    Logger.debug("Funding user #{Encoding.to_hex(receiver.addr)} with #{amount} from utxo: #{Utxo.pos(utxo)}")
+    {:ok, blknum, txindex} = Transaction.submit_tx([utxo], outputs, [faucet_account], @fund_child_chain_account_retries)
 
-    updated_state = Map.put(state, :utxos, %{state.utxos | token => {change_utxo, change}})
+    next_faucet_utxo = %Utxo{blknum: blknum, txindex: txindex, oindex: 0, amount: change}
+    user_utxo = %Utxo{blknum: blknum, txindex: txindex, oindex: 1, amount: amount}
+
+    updated_state = Map.put(state, :utxos, Map.put(state.utxos, currency, next_faucet_utxo))
 
     {:reply, {:ok, user_utxo}, updated_state}
   end
 
-  defp get_funded_faucet_account(opts) do
-    faucet_private_key = Keyword.fetch!(opts, :faucet_private_key)
-    {:ok, faucet} = Account.new(faucet_private_key)
+  @spec check_sufficient_funds(state(), Utxo.address_binary(), pos_integer()) :: Utxo.t()
+  defp check_sufficient_funds(
+         %{utxos: utxos, faucet_account: faucet_account, fee: fee},
+         currency,
+         amount
+       ) do
+    utxo =
+      case Map.has_key?(utxos, currency) do
+        true ->
+          utxos[currency]
+
+        _ ->
+          faucet_account.addr
+          |> get_utxos()
+          |> get_largest_utxo_by_currency(faucet_account, currency)
+          |> get_largest_utxo_by_currency(faucet_account, currency)
+      end
+
+    case utxo.amount - amount - fee < 0 do
+      true -> deposit(faucet_account, currency)
+      _ -> utxo
+    end
+  end
+
+  @spec get_utxos(Utxo.address_binary()) :: list()
+  defp get_utxos(address) do
+    {:ok, response} =
+      Api.Account.account_get_utxos(
+        Connection.client(),
+        %Model.AddressBodySchema1{
+          address: Encoding.to_hex(address)
+        }
+      )
+
+    Jason.decode!(response.body)["data"]
+  end
+
+  @spec get_largest_utxo_by_currency(list(), Account.t(), Utxo.address_binary()) :: Utxo.t()
+  defp get_largest_utxo_by_currency([], faucet_account, currency), do: deposit(faucet_account, currency)
+
+  defp get_largest_utxo_by_currency(utxos, faucet_account, currency) do
+    utxos = Enum.filter(utxos, fn utxo -> currency == LoadTest.Utils.Encoding.from_hex(utxo["currency"]) end)
+
+    case length(utxos) do
+      0 -> deposit(faucet_account, currency)
+      _ -> get_largest_utxo(utxos, currency)
+    end
+  end
+
+  @spec get_largest_utxo(list(), Utxo.address_binary()) :: Utxo.t()
+  defp get_largest_utxo(utxos, _currency) do
+    utxo = Enum.max_by(utxos, fn x -> x["amount"] end)
```

Comment:
The second argument can be shortened to:
`&(&1["amount"])`
the `&1` is a shorthand for first argument of a function